bookkeeper-server/src/main/java/org/apache/bookkeeper/conf/ServerConfiguration.java

```diff
@@ -1101,6 +1105,101 @@ public ServerConfiguration setMajorCompactionThreshold(double threshold) {
         return this;
     }
 
+    /**
+     * Get threshold of median major compaction.
+     *
+     * <p>same with the @see #getMajorCompactionThreshold()
+     * the high threshold is used when the system is on a low load status
+     *
+     * @return threshold of median major compaction
+     */
+    public double getMedianMajorCompactionThreshold() {
+        return getDouble(MAJOR_MEDIAN_COMPACTION_THRESHOLD, 0.7f);
+    }
+
+    /**
+     * Set threshold of median major compaction.
+     *
+     * @see #getMedianMajorCompactionThreshold()
+     *
+     * @param threshold
+     *          Threshold of median major compaction
+     * @return server configuration
+     */
+    public ServerConfiguration setMedianMajorCompactionThreshold(double threshold) {
+        setProperty(MAJOR_MEDIAN_COMPACTION_THRESHOLD, threshold);
+        return this;
+    }
+
+    /**
+     * Get threshold of high major compaction.
+     *
+     * <p>same with the @see #getMajorCompactionThreshold()
+     * the high threshold is used when the system is on a very low load status
+     *
+     * @return threshold of high major compaction
+     */
+    public double getHighMajorCompactionThreshold() {
+        return getDouble(MAJOR_HIGH_COMPACTION_THRESHOLD, 0.9f);
+    }
+
+    /**
+     * Set threshold of high major compaction.
+     *
+     * @see #getHighMajorCompactionThreshold()
+     *
+     * @param threshold
+     *          Threshold of major compaction
+     * @return server configuration
+     */
+    public ServerConfiguration setHighMajorCompactionThreshold(double threshold) {
+        setProperty(MAJOR_HIGH_COMPACTION_THRESHOLD, threshold);
+        return this;
+    }
+
+    /**
+     * Get median major compaction cron expression.
+     * default is 0-3 h every day.
+     */
+    public String getMedianMajorCompactionCron() {
+        return getString(MEDIAN_MAJOR_COMPACTION_CRON, "0 * 0-3 ? * *");
```

Comment:
I think default value should be returning null. if the return value is null, the cron based compaction is disabled.